### DISCUSSION TITLE
Why not implement Brave patches?

### DISCUSSION THREAD

**Discussion by peternrdstrm** [bromite/bromite#1245] [Ideas]
I have been looking for a "perfect" mobile browser for a long time. And after testing every single open source browser I could find, Bromite and Brave were the last two standing. Two browsers very similar, yet different in key areas. The "perfect" browser would be a mix of the two (I'll go into this later). 

So I came to the conclusion, that some Brave patches would really help Bromite out. 
I think, that it's pre-set user agent and randomized timezone are great things, but I miss a stronger ad blocker and better fingerprinting protection. Not to mention the many convinience features Brave has. 
I think that Bromite is still superior, as it takes privacy much more seriously, and it's security seems to be better, especially with the JITless switch now. 

So here are the patches/features that would massively improve Bromite in my opinion. 

1. The strict ad-blocker. 
Having a strong adblocker that blocks every ad and tracker there is, is a priority IMO. Yes, it would break some sites, but it would also imrove loading speeds, data consumption and battery life. It also wouldn't be very hard to implement a downgrade/bypass switch into the site setting menu. 
2. Having a stricter fingerprint randomization patch.
Panopticlick seems to suggest, that Brave randomizes your fingerprint, but Bromite doesn't. Altough, on other sites, it is clear, that Bromite does randomize some data. So what if,it would do it on a larger scale. Maybe randomize more data. 
3. Having the HTTPS Upgrade patch
This is straight forward. I actually asked for this a while ago, I'm sure they closed it.
4. Convinience features : 
As of my knowleadge of chromium based android browsers, Brave is one of the only ones, that let you decide, weather you want a grid or a normal card tab layout. I really, and I mean REALLY hate the grid layout, and would love to see it reimplemented into Bromite. 
The other one is the bottom toolbar. It's just soooo much easier to user, on large phones (mine's 6.55")

Now, @csagan5 to adress this, yes, I know, that it's much harder to do all this, than to write it down, and you might not feel right about using Brave patches, but overtime, implementing theese features would be the key to Bromite's succes. As I said, Bromite is the closest, I've got to the perfect android browser. And I want you to win.  Brave, IMO is bloated, and has a much more "mainstream apeal first, privacy second" approach, while you, and Bromite are focused, and comitted to bringing users the privacy they deserve. And I'm very thankfull that you do this for us. 
I might not be skikled enough to help with the development, but for everything else, I could, and would happily help you. Translations, testing, etc. As a content creator, I will probabily cover Bromite in the future too.  But hey that's it for now....or maybe not.

One more thing jumped to my mind. I have been testing browsers with browseraudit.com
And there are two things, that I'm concerned about :
Enableing #strict-origin-isolation adds 6 critical issues. As this is a security feature, that should increase security, I don't think this should be the case.
By default, (maybe only in incognito-only, not sure about that one) the critical issue, "Cookie set by javascript trough HTTP" appears . I have no idea what's going on, just wanted to throw theese two in here

**Comment by uazo** [comment]
regarding BrowserAudit

> Enabling #strict-origin-isolation adds 6 critical issues. As this is a security feature, that should increase security, I don't think this should be the case.

#strict-origin-isolation: `Experimental security mode that strengthens the site isolation policy. Controls whether site isolation should use origins instead of scheme and eTLD+1. – Mac, Windows, Linux, Chrome OS, Android`

no security problems, indeed, an improvement. 
that check detects a difference in the implementation of the standards (see source codes https://github.com/browseraudit/), and it is correct that it is detected: the test checks for allowed access permission to the dom, that flag blocks it

> Cookie set by javascript trough HTTP

I think this is a bug of the site, it should be checked with the standards, IMO it is correct that cookies via javascript go through http (note http not https).

**Comment by csagan5** [comment]
> Now, @csagan5 to adress this, yes, I know, that it's much harder to do all this, than to write it down, and you might not feel right about using Brave patches

There is no problem at using Brave's patches, if the license allows it. Feel free to start submitting those you think might be an improvement for Bromite, they will go through the review process as any other patch.

> but overtime, implementing these features would be the key to Bromite's success.

Little misunderstanding here: this project is not pursuing any success, if you like it - use it, contribute to it, share it, otherwise don't.
